docs/source/public_api/kernels.rst

```diff
@@ -0,0 +1,12 @@
+probnum.kernels
+========================
+
+Kernels or covariance functions.
```

Comment:
Sorry if this is late, but I think now is still better than in the future: 

What are your thoughts on changing `Kernel` to `CovarianceKernel`. The term "kernel" is associated with so many things. (Same for "covariance" btw.). I personally am never sure whether I prefer "kernel" or "covariance function" so I tend to use both. This would make the names a bit longer, but would avoid any sort of confusion... I guess?

Reply:
Long term this will move into `probnum.prob.kernels` which I think will alleviate this confusion. To me `CovarianceKernel` sounds somewhat doubled, alternatively I could also imagine `CovarianceFunction`.